File: 386-lang/homework1/wordcount.py
import fileinput, string

#function partially from: http://stackoverflow.com/questions/1342000/how-to-make-the-python-interpreter-correctly-handle-non-ascii-characters-in-stri
def removeNonAscii(s): 
	s = s.translate(None, string.punctuation)
	return "".join(i for i in s if ord(i)<128)

story = ""
f = fileinput.input()

# Construct single string out of story
# Lowercase, remove non-ascii chars, returns & punctuation
for line in f:
	line = line.lower()
	line = removeNonAscii(line)
	story += line[:-1] + " "

#parse story by spaces, alphabetize & sort unique words
story = sorted(story.split())
unique = set(story)
unique = sorted(unique)

#print occurances of individual words
for s in unique:
	print s + " " + str(story.count(s))

File: 386-lang/homework2/wordcount.js
//regex from: http://stackoverflow.com/questions/4328500/how-can-i-strip-all-punctuation-from-a-string-in-javascript-using-regex

var readline = require('readline');
var stream = require('stream');

var story = "";
var words = [];
var dummyOut = new stream;
var rl = readline.createInterface(process.stdin, dummyOut);

//for each line
rl.on('line', function(line) {

	line = line.toLowerCase();

	//replaces all non-ascii and punctuation
	line = line.replace(/[^\w\s]|_/gm, "")
	story += line + " "


});

rl.on('close', function() {
	words = story.split(" ");
	var unique = [];
	

	//Filtering out unique elements & straggling spaces
	for (var i = 0; i < words.length; i++) {
		if (unique.indexOf(words[i]) == -1 && words[i] != "") {
			unique.push(words[i]);
		}
	}

	unique.sort();

	//Printing occurances
	for (var i = 0; i < unique.length; i++) {
		var count = 0;
		for (var j = 0; j < words.length; j++) {
			if (unique[i] == words[j]){
				count++;
			}
		}
		console.log(unique[i] + " " + count);
	}

});